truckersmp_cli/__init__.py

```diff
@@ -718,59 +725,110 @@ def get_supported_game_versions():
 
 
 def update_game():
-    """Update game and Proton."""
-    # make sure steam is closed before updating
-    # it's possible to update with the steam client open but the client looses
-    # all connectivity and asks for password and steam guard code after restart
-    try:
-        subproc.check_call(
-          ["pgrep", "-u", getuser(), "-x", "steam"], stdout=subproc.DEVNULL)
-        logging.debug("Closing Steam")
-        subproc.call(["steam", "-shutdown"])
-    except Exception:
-        pass
+    """
+    Update game and Proton via SteamCMD.
+
+    We make sure Steam is closed before updating.
+    It's possible to update with the Steam client open but the client looses
+    all connectivity and asks for password and Steam Guard code after restart.
 
-    if not os.path.isdir(args.gamedir):
-        logging.debug("Creating directory {}".format(args.gamedir))
-        os.makedirs(args.gamedir, exist_ok=True)
+    When "--wine" is specified, this function retrieves/uses Windows version of
+    SteamCMD. When "--proton" is specified, this retrieves/uses
+    Linux version of SteamCMD.
+    """
+    env = os.environ.copy()
+    steamcmd_prolog = ""
+    steamcmd_cmd = []
+    if args.proton:
+        # we don't use system SteamCMD because something goes wrong in some cases
+        # see https://github.com/lhark/truckersmp-cli/issues/43
+        steamcmd = os.path.join(Dir.steamcmddir, "steamcmd.sh")
+        steamcmd_url = URL.steamcmdlnx
+        gamedir = args.gamedir
+    else:
+        wine = os.environ["WINE"] if "WINE" in os.environ else "wine"
+        env["WINEDEBUG"] = "-all"
+        env["WINEARCH"] = "win64"
+        env_steam = env.copy()
+        env_steam["WINEPREFIX"] = args.prefixdir
+        # use a prefix only for SteamCMD to avoid every-time authentication
+        env["WINEPREFIX"] = Dir.steamcmdpfx
+        # don't show "The Wine configuration is being updated" dialog
+        # or install Gecko/Mono
+        env["WINEDLLOVERRIDES"] = "winex11.drv="
```

Comment:
This is overwriting all user settings for this. We should append our things, see https://github.com/ValveSoftware/Proton/blob/proton_5.0/proton#L557

Besides that this didn't worked at all. I still got all the messages about updating and installing Gecko. A quick search shows me I should use `mscoree,mshtml=` but that also didn't work.

Reply:
The first point is still something to discuss: I have set dll overrides system wide and this line will overwrite my setting if I'm not mistaken.

Reply:
We force to replace `WINEDLLOVERRIDES` when:
* executing `wine winedbg --command "info process"` in `check_steam_process()`
* executing `wine --version` in `update_game()`
* executing `winepath -w` in `update_game()`
* running `steamcmd.exe` in `update_game()`
I think `WINEDLLOVERRIDES` should not be able to be changed by user when doing these tasks. For example, `steamcmd.exe` does nothing when `WINEDLLOVERRIDES=ws2_32=` is specified by user.

---

When running game, `start_with_wine()` may change `WINEDLLOVERRIDES`: PR #94 fixes this.
```python
    env["WINEDLLOVERRIDES"] = "d3d11=;dxgi=" if not args.enable_d3d11 else ""
```

Reply:
agreed